File: solutions/scala/largest-series-product/1/src/main/scala/LargestSeriesProduct.scala
object Series {
  def largestProduct(span: Int, digits: String): Option[Int] = {
    if (span > digits.length || span < 0) None
    else if (digits.map(_.asDigit).count(x => x < 10 && x >= 0) != digits.length) None
    else (span, digits) match {
      case (0, _) => Some(1)
      case (_, "") => None
      case (_, _) => Some(digits.map(_.asDigit).sliding(span).map(_.product).max)
    }
  }
}

File: solutions/lua/largest-series-product/1/largest-series-product.lua
return function(config)
  local max = 0
  if #config.digits < config.span or config.span < 0 then error('Invalid') end
  for i = 1, #config.digits - config.span + 1 do
    local match, number = config.digits:sub(i, i + config.span - 1), 1
    for j = 1, #match do number = number * match:sub(j, j) end
    if number > max then max = number end
  end
  return max
end
